(SERVER-2822) Doc updates for multithreaded mode (#2360)
puppetlabs/puppetserver: documentation/config_file_puppetserver.markdown

    * `master-log-dir`: Optional. The path to the log directory. If nothing is specified, it uses the Puppet default `/var/log/puppetlabs/puppetserver`.

>     * `max-active-instances`: Optional. The maximum number of JRuby instances allowed. The default is 'num-cpus - 1', with a minimum value of 1 and a maximum value of 4. In multithreaded mode, this control the number of threads allowed to run concurrently through the single JRuby instance.

Reviewer: ```suggestion
    * `max-active-instances`: Optional. The maximum number of JRuby instances allowed. The default is 'num-cpus - 1', with a minimum value of 1 and a maximum value of 4. In multithreaded mode, this controls the number of threads allowed to run concurrently through the single JRuby instance.
```

        JRuby flushing can be useful for working around buggy module code that would otherwise cause memory leaks, but it slightly reduces performance whenever a new JRuby instance reloads all of the Puppet Ruby code. If memory leaks from module code are not an issue in your deployment, the default value of 0 performs best.

>     * `multithreaded`: Optional, false by default. Configures Puppet Server to use a single JRuby instance to process requests that require a JRuby, processing a number of threads up to `max-active-instances` at a time. Reduces the memory footprint of the server by only requiring a single JRuby. When this setting is configured, Puppet's `multithreaded` setting must also be set to `true`.

Reviewer: Should we add a note to `max-active-instances` about what it means when `multithreaded` is on?

Reviewer: I still think we should do this.

Author: Oh yeah, I think this comment got hidden, sorry.

>     * `multithreaded`: Optional, false by default. Configures Puppet Server to use a single JRuby instance to process requests that require a JRuby, processing a number of threads up to `max-active-instances` at a time. Reduces the memory footprint of the server by only requiring a single JRuby. When this setting is configured, Puppet's `multithreaded` setting must also be set to `true`.

Reviewer 2: ```suggestion
    * `multithreaded`: Optional, false by default. Configures Puppet Server to use a single JRuby instance to process requests that require a JRuby, processing a number of threads up to `max-active-instances` at a time. Reduces the memory footprint of the server by only requiring a single JRuby. When you configure this setting, you must also set Puppet's `multithreaded` setting to `true`.
```

>     * `multithreaded`: Optional, false by default. Configures Puppet Server to use a single JRuby instance to process requests that require a JRuby, processing a number of threads up to `max-active-instances` at a time. Reduces the memory footprint of the server by only requiring a single JRuby. When you configure this setting, you must also set Puppet's `multithreaded` setting to `true`.

Reviewer 3: I don't remember putting a `multithreaded` setting into Puppet. Where is that?

Author: Oh, maybe we didn't? Oops. I guess we're mostly just checking if we run in JRuby?

Reviewer 3: Yeah, everything should just use the threadsafe implementation if it's running on java

Author: Fixed. Thanks for catching that.
